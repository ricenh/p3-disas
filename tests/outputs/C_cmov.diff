--- outputs/C_cmov.txt	2023-11-01 19:33:16.990485047 -0400
+++ expected/C_cmov.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,11 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of executable contents:
+  0x100:                               | .pos 0x100 code
+  0x100:                               | _start:
+  0x100: 20 01                         |   rrmovq %rax, %rcx
+  0x102: 21 23                         |   cmovle %rdx, %rbx
+  0x104: 22 45                         |   cmovl %rsp, %rbp
+  0x106: 23 67                         |   cmove %rsi, %rdi
+  0x108: 24 89                         |   cmovne %r8, %r9
+  0x10a: 25 ab                         |   cmovge %r10, %r11
+  0x10c: 26 ce                         |   cmovg %r12, %r14
+
