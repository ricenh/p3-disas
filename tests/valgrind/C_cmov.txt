==1335428== Memcheck, a memory error detector
==1335428== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1335428== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1335428== Command: ../y86 -d inputs/cmov.o
==1335428== 
../y86: invalid option -- 'd'
Usage: ../y86 <option(s)> mini-elf-file
 Options are:
  -h      Display usage
  -H      Show the Mini-ELF header
  -a      Show all with brief memory
  -f      Show all with full memory
  -s      Show the program headers
  -m      Show the memory contents (brief)
  -M      Show the memory contents (full)
  -d      Disassemble code contents
  -D      Disassemble data contents
==1335428== 
==1335428== HEAP SUMMARY:
==1335428==     in use at exit: 0 bytes in 0 blocks
==1335428==   total heap usage: 1 allocs, 1 frees, 8,192 bytes allocated
==1335428== 
==1335428== All heap blocks were freed -- no leaks are possible
==1335428== 
==1335428== For lists of detected and suppressed errors, rerun with: -s
==1335428== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
