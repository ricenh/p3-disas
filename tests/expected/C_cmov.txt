Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 20 01                         |   rrmovq %rax, %rcx
  0x102: 21 23                         |   cmovle %rdx, %rbx
  0x104: 22 45                         |   cmovl %rsp, %rbp
  0x106: 23 67                         |   cmove %rsi, %rdi
  0x108: 24 89                         |   cmovne %r8, %r9
  0x10a: 25 ab                         |   cmovge %r10, %r11
  0x10c: 26 ce                         |   cmovg %r12, %r14

